internal/cmd/launcher_test.go

```diff
@@ -348,3 +350,60 @@ func TestIsAnalysisRequired(t *testing.T) {
 		})
 	}
 }
+
+func TestGetBuildServiceURL(t *testing.T) {
+	t.Parallel()
+
+	testCases := []struct {
+		name                      string
+		buildSrvURL               string
+		enableCommunityExtensions bool
+		expectErr                 bool
+		expected                  string
+	}{
+		{
+			name:                      "default build service url",
+			buildSrvURL:               "https://build.srv",
+			enableCommunityExtensions: false,
+			expectErr:                 false,
+			expected:                  "https://build.srv/cloud",
+		},
+		{
+			name:                      "enable community extensions",
+			buildSrvURL:               "https://build.srv",
+			enableCommunityExtensions: true,
+			expectErr:                 false,
+			expected:                  "https://build.srv/oss",
+		},
+		{
+			name:                      "invalid buildServiceURL",
+			buildSrvURL:               "https://host:port",
+			enableCommunityExtensions: false,
+			expectErr:                 true,
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			t.Parallel()
+
+			logger := &logrus.Logger{ //nolint:forbidigo
+				Out: io.Discard,
+			}
+
+			flags := state.GlobalFlags{
+				BinaryProvisioning:        true,
+				BuildServiceURL:           tc.buildSrvURL,
+				EnableCommunityExtensions: tc.enableCommunityExtensions,
+			}
+
+			buildSrvURL, err := getBuildServiceURL(flags, logger)
+			if tc.expectErr {
+				assert.Error(t, err)
+			} else {
+				assert.NoError(t, err)
+				assert.Equal(t, tc.expected, buildSrvURL)
+			}
```

Comment:
```suggestion
		require.Equal(t, tc.expectErr, err)
		assert.Equal(t, tc.expectURL, buildSrvURL)
```
This could be simplified like this